app-creator/client/src/utils/template-generation.ts

```diff
@@ -134,3 +140,114 @@ export function getWidgetElement({
 
   return { element, dropzone, map, draggable };
 }
+
+/**
+ * Retrieves previous template from localStorage and populate new template with widgets.
+ */
+export function transferData() {
+  try {
```

Comment:
It's best to keep a `try` clause small, so only wrap around the place where an error might be happening.